modules/eks/submodules/privatelink/vpc-endpoint-services.tf

```
@@ -37,13 +37,14 @@ resource "aws_lb" "nlbs" {
 resource "aws_lb_target_group" "target_groups" {
   for_each = { for entry in local.listeners : "${entry.service}.${entry.port}" => entry }
 
-  name     = substr("${var.deploy_id}-${each.value.service}-${each.value.port}", -32, -1)
+  name     = "${var.deploy_id}-${substr(each.value.service,0,10)}-${each.value.port}"
```

```suggestion
  name     = "${var.deploy_id}-${substr(each.value.service,0,9)}-${each.value.port}"
```
` (16 deploy_id + 5 port+ 2 - + 9 service)`